### PR TITLE
docs: update environment instructions

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: bluebrain/documentation/installing_with_environments.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ and then activating it, and fixing the configuration to consider all
 software together:
 
     $ spack env activate -d spamhameggs
-    $ spack config add config:spack:concretization=together
+    $ spack config add concretizer:unify:true
```
Comment (cattabiani) [inline bluebrain/documentation/installing_with_environments.md:21]: I would put this after `create`. After `activate` looks like you need it every time you activate the env

Comment (heerener) [reply bluebrain/documentation/installing_with_environments.md:21]: If you want the setting to apply to your environment, you need to activate it first. If you don't activate it first, the setting will go to `~/.spack/concretizer.yaml`, only to be overridden by your environment config.

Comment (cattabiani) [reply bluebrain/documentation/installing_with_environments.md:21]: oh, that makes sense!

Comment (matz-e) [reply bluebrain/documentation/installing_with_environments.md:21]: Ah, I see. Actually, it is needed after the _first_ activation. For future activations, it's already saved in the environment.

Comment (cattabiani) [reply bluebrain/documentation/installing_with_environments.md:21]: Maybe 1 sentence to specify this. It was not clear to me

Comment (matz-e) [reply bluebrain/documentation/installing_with_environments.md:21]: OK, will add! Thanks for the feedback!

**Review by heerener — DISMISSED** [review]

**Review by cattabiani — APPROVED** [review]
LGTM!

**Pull request merged**